trace-utils/src/trace_utils.rs

```diff
@@ -526,6 +525,30 @@ pub fn enrich_span_with_mini_agent_metadata(
     }
 }
 
+pub fn enrich_span_with_google_cloud_function_metadata(
```

Comment:
If we're adding a new method to add metadata specifically for google cloud functions can you move the code for setting the `project_id` and `location`?

https://github.com/DataDog/libdatadog/blob/373b7783730a8191bfb6a1f0135791782cdcc22d/trace-utils/src/trace_utils.rs#L513-L519

That should also allow for using the source data directly to construct the resource name instead of setting the span metadata in one function then fetching it again in another.

Reply:
Thanks for the comment. I added it